### ISSUE TITLE
DocFX with Unity - comments in MarkDown file but not in the DocFX generated documentation

### ISSUE THREAD

**Issue by bertrandArnaud** [dotnet/docfx#3066]
**DocFX Version Used**: v2.36.2 - Nuget Package

**Template used**: (`default`)

Hello,

I'm working on an Unity project and wanted to make my documentation using DocFX. The fact is DocFX generates the doc only for an imported Asset.

I tried to understand why and it looks like Unity scripts aren't in a namespace. So i created a test class inside a namespace with some random XML comments. The result was the same, the created test class wasn't in the DocFX documentation.

It's important to note that all my comments are in the C# MarkDown file.

Thanks in advance.

**Comment by herohua** [comment]
@vwxyzh @vicancy is this supported in current version of DocFX?

FYI @TianqiZhang @superyyrrzz

**Comment by vicancy** [comment]
Hi @bertrandArnaud did you run `docfx metadata` against the source file? Any logs after running that?

**Comment by bertrandArnaud** [comment]
Hello @vicancy,

I ran docfx metadata and it finished with 103 warnings and 0 error. I cleaned my projects a bit, deleting old .csproj files. Rebuilded and tried again : suceeded, 0 warning, 0 error but still no documentation for my classes.

After many tries and comparaisons I finished to understand the reason why docfx was referencing only certains classes : 
1. The class must be referenced public.
2. The class must be inside a namespace.

It's a bit new for me since I only used Doxygen in past to generate PHP documentation and I don't remember of these conditions to have a class referenced.

Maybe it could be explicited on the DocFX, it may not be explicit for new users. 

Thanks for your help !
